pika/channel.py

```diff
@@ -983,15 +1105,16 @@ def _on_deliver(self, method_frame, header_frame, body):
 
         """
         consumer_tag = method_frame.method.consumer_tag
+
         if consumer_tag in self._cancelled:
             if self.is_open and consumer_tag not in self._consumers_with_noack:
                 self.basic_reject(method_frame.method.delivery_tag)
             return
+
         if consumer_tag not in self._consumers:
-            return self._add_pending_msg(consumer_tag, method_frame,
-                                         header_frame, body)
-        while self._pending[consumer_tag]:
```

Comment:
REVIEWME removed "pending deliveries" support that was in support of legacy `BlockingConnection` and is no longer needed.